app/controllers/locationOfGoods/IdentificationController.scala

```diff
@@ -39,50 +39,58 @@ class IdentificationController @Inject() (
-        case identifiers =>
-          val preparedForm = request.userAnswers.get(IdentificationPage) match {
-            case None        => form
-            case Some(value) => form.fill(value)
-          }
+  def onPageLoad(lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions
+    .requireData(lrn)
+    .andThen(getMandatoryPage(LocationTypePage))
+    .async {
+      implicit request =>
+        val locationType = request.arg
+        locationType match {
+          case LocationType("B", _) =>
+            IdentificationPage.writeToUserAnswers(LocationOfGoodsIdentification("Y", "Authorisation number"))
+            Future.successful(Redirect(controllers.locationOfGoods.routes.AuthorisationNumberController.onPageLoad(lrn, mode)))
 
-          Future.successful(Ok(view(preparedForm, lrn, identifiers, mode)))
-      }
-  }
+          case _ =>
+            locationOfGoodsIdentificationTypeService.getLocationOfGoodsIdentificationTypes(request.userAnswers).map {
+              locationOfGoodsIdentificationTypes =>
+                val preparedForm = request.userAnswers.get(IdentificationPage) match {
+                  case None        => form(locationOfGoodsIdentificationTypes)
+                  case Some(value) => form(locationOfGoodsIdentificationTypes).fill(value)
+                }
+                Ok(view(preparedForm, lrn, locationOfGoodsIdentificationTypes, mode))
+            }
+        }
+    }
 
   def onSubmit(lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions.requireData(lrn).async {
     implicit request =>
-      form
-        .bindFromRequest()
-        .fold(
-          formWithErrors => Future.successful(BadRequest(view(formWithErrors, lrn, LocationOfGoodsIdentification.values(request.userAnswers), mode))),
-          value => redirect(mode, IdentificationPage, value)
-        )
-  }
-
-  private def redirect(
-    mode: Mode,
-    page: QuestionPage[LocationOfGoodsIdentification],
-    value: LocationOfGoodsIdentification
-  )(implicit request: DataRequest[_]): Future[Result] = {
-    implicit val navigator: UserAnswersNavigator = navigatorProvider(mode)
-    page
-      .writeToUserAnswers(value)
-      .updateTask()
-      .writeToSession()
-      .navigate()
+      locationOfGoodsIdentificationTypeService.getLocationOfGoodsIdentificationTypes(request.userAnswers).flatMap {
+        locationOfGoodsIdentificationTypes =>
+          form(locationOfGoodsIdentificationTypes)
+            .bindFromRequest()
+            .fold(
+              formWithErrors => Future.successful(BadRequest(view(formWithErrors, lrn, locationOfGoodsIdentificationTypes, mode))),
+              value => {
+                implicit val navigator: UserAnswersNavigator = navigatorProvider(mode)
+                IdentificationPage
+                  .writeToUserAnswers(value)
+                  .updateTask()
+                  .writeToSession()
+                  .navigate()
+              }
```

Comment:
Reintroduce the `redirect` method and change this back to `value => redirect(mode, IdentificationPage, value)`

Reply:
@kenneth-rayner this is still outstanding